com.unity.visualeffectgraph/Editor/Compiler/VFXCodeGenerator.cs

```diff
@@ -556,6 +741,40 @@ static private StringBuilder Build(VFXContext context, string templatePath, VFXC
             return stringBuilder;
         }
 
+        private static StringBuilder TryBuildFromShaderGraph(VFXShaderGraphParticleOutput context, VFXContextCompiledData contextData)
+        {
+            var stringBuilder = new StringBuilder();
+
+            // Reconstruct the ShaderGraph.
+            var path = AssetDatabase.GetAssetPath(context.shaderGraph);
+
+            List<PropertyCollector.TextureInfo> configuredTextures;
+            AssetCollection assetCollection = new AssetCollection();
+            MinimalGraphData.GatherMinimalDependenciesFromFile(path, assetCollection);
+
+            var textGraph = File.ReadAllText(path, Encoding.UTF8);
+            var graph = new GraphData
+            {
+                messageManager = new MessageManager(), assetGuid = AssetDatabase.AssetPathToGUID(path)
+            };
+            MultiJson.Deserialize(graph, textGraph);
+            graph.OnEnable();
+            graph.ValidateGraph();
+
+            // Configure the state of VFX target generation utils.
+            using (new VFXSubTarget.CompilationScope(context, contextData))
```

Comment:
Configure the static state of the VFXSubTarget with the context and context data, before invoking the shader graph generator.

Reply:
Hax!   :)

Instead of storing the replacement sub target info in a global variable...
Could we just modify the graph itself here, removing the old subtarget and adding our replacement?

Or, I think ideally, we make a version of GetShaderText that takes a replacement target (or target array?) -- when called with a non-null replacement target it uses that instead of the targets embedded in the graph.  That would let us replace the target more directly here.